---
title: "Venn Diagram"
author: "Allison Andre"
date: "2/24/2021"
output: 
  html_document: 
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load the libraries
```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(janitor)
library(here)
library(naniar)
library(ggvenn)
library(VennDiagram)
``` 

## Load the Data
```{r}
tidier_microbiome <- readr::read_csv("/Users/allyandre/Desktop/BIS15L_Project_Group11/tidy_data/tidiermicrobiome.csv")
tidier_microbiome
```
```{r}
asd_tidier <- readr::read_csv("/Users/allyandre/Desktop/BIS15L_Project_Group11/tidy_data/asd_group_tidier.csv")
asd_tidier
```

```{r}
td_tidier <- readr::read_csv("/Users/allyandre/Desktop/BIS15L_Project_Group11/tidy_data/td_group_tidier.csv")
td_tidier
```

## Prepare data for Venn diagram
```{r}
td_venn <-
  td_tidier %>% 
  rename(genus_td = "genus") %>% 
  filter(abundance != 0) %>% 
  select(genus_td, patient_id)
```

```{r}
asd_venn <-
  asd_tidier %>% 
  rename(genus_asd = "genus") %>% 
  filter(abundance != 0)
```

```{r}
both_venn <- full_join(td_venn, asd_venn, by = "patient_id")
both_venn
```

```{r}
both_venn$genus_td <- as.factor(both_venn$genus_td)
both_venn$genus_asd <- as.factor(both_venn$genus_asd)
```

```{r}
both_venn2 <- 
  both_venn %>% 
  filter(genus_td != "NA" | genus_asd != "NA")
```


## Build Venn Diagram

```{r}
venn_data <- list(
  "TD Genera" = both_venn2$genus_td,
  "ASD Genera" = both_venn2$genus_asd
)
```



```{r}
ggvenn(venn_data, fill_color = c("yellow", "turquoise1"))
```

## Try to build a COOLER (aka to-scale) venn diagram

```{r}
anyNA(venn_data)
```


```{r}
grid.newpage()
draw.pairwise.venn(area1 = 1264,                      
                   area2 = 1101,
                   cross.area = 1068, 
                   fill = c("yellow", "turquoise1"), 
                   category = c("TD Genera", "ASD Genera"))
```


## Bargraphs of number of genera unique to each group

```{r}
unique_names <- c("TD group", "ASD Group")
unique_counts <- c(196, 33)
unique_genera <- data_frame(unique_names, unique_counts)
unique_genera
```

```{r}
unique_genera %>% 
  ggplot(aes(x = unique_names, y = unique_counts, fill = unique_names)) +
  geom_col(color = "black") + 
  scale_fill_manual(values = c("turquoise1", "yellow")) +
  theme_linedraw() +
  labs(title = "Number of Unique Genera Per Group",
       x = NULL,
       y = "Unique Genera") +
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5)) 
```

#Prep data for Venn Diagram by Gender
##Capitalize A and B
```{r}
both_venn$condition <- str_to_upper(both_venn$condition, locale = "en")
```

## Wrangle the data so they have enough in common to join them together
### Abundance Data
```{r}
asd <- readr::read_csv("ASD_meta_abundance_2.csv")
asd
```
```{r}
asd2 <- 
  asd %>% 
  pivot_longer(-Taxonomy, 
               names_to = "subject", 
               values_to = "abundance") %>% 
  filter(str_detect(Taxonomy, "Unclassified") == FALSE) %>% 
  separate(Taxonomy, into = c("genus", "species"), sep = ";") %>% 
  separate(genus, into = c("g", "genus"), sep = "__") %>% 
  separate(species, into = c("s", "species"), sep = "__") %>% 
  select( -g, -s, -species) %>% 
  rename(sample_id = "subject")
asd2
```

###Demographics data
```{r}
demographics <- readr::read_csv("/Users/allyandre/Desktop/BIS15L_Project_Group11/untidy_data/demographics.csv")
demographics
```

```{r}
demographics2 <-
  demographics %>% 
  clean_names() %>% 
  filter(metagenomic_sequencing == "Yes") %>% 
  select(sample_id, gender, age)
demographics2
```

##Join the two dataframes
```{r}
demo_venn <- full_join(asd2, demographics2, by = "sample_id") %>% 
  select(-c(sample_id, abundance, age))
demo_venn

demo_abundance <- full_join(asd2, demographics2, by = "sample_id")
```

```{r}
write.csv(demo_abundance, file = "demographics_and_abundance.csv", row.names = FALSE)
```


##Divide the genera by gender
```{r}
demo_venn_f <- 
  demo_venn %>% 
  filter(gender == "Female") %>% 
  rename(genus_f = "genus")
demo_venn_f
demo_venn_m <-
  demo_venn %>% 
  filter(gender == "Male") %>% 
  rename(genus_m = "genus")
demo_venn_m
```

```{r}
demo_venn_f$genus_f <- as.factor(demo_venn_f$genus_f)
demo_venn_m$genus_m <- as.factor(demo_venn_m$genus_m)
```

#Venn diagrams of overlap by gender
##Not-to=Scale Venn Diagram
```{r}
demo_venn_data <- list(
  "Male Genera" = demo_venn_m$genus_m,
  "Female Genera" = demo_venn_f$genus_f
)
```

```{r}
ggvenn(demo_venn_data, fill_color = c("yellow", "turquoise1"))
```
## To-Scale Venn Diagram
```{r}
grid.newpage()
draw.pairwise.venn(area1 = 1335,                      
                   area2 = 1335,
                   cross.area = 1335, 
                   fill = c("yellow", "turquoise1"), 
                   category = c("Male Genera", "Female Genera"))
```
## Venn Diagram by Age Group
```{r}
age_venn <- full_join(asd2, demographics2, by = "sample_id") %>% 
  select(-c(sample_id, abundance, gender)) %>% 
  mutate(age = case_when(age <= 7 ~ "younger",
                         age > 7 ~ "older"))
age_venn$genus <- as.factor(age_venn$genus)
age_venn_y <-
  age_venn %>% 
  filter(age == "younger") %>% 
  rename(genus_y = "genus")

age_venn_o <-
  age_venn %>% 
  filter(age == "older") %>% 
  rename(genus_o = "genus")
```

```{r}
age_venn_data <- list(
  "Younger Genera" = age_venn_y$genus_y,
  "Older Genera" = age_venn_o$genus_o
)
```

```{r}
ggvenn(age_venn_data, fill_color = c("yellow", "turquoise1"))
```


